saleor/graphql/product/types/products.py

```diff
@@ -307,8 +312,11 @@ def resolve_digital_content(root: models.ProductVariant, *_args):
         return getattr(root, "digital_content", None)
 
     @staticmethod
-    def resolve_stock_quantity(root: models.ProductVariant, *_args):
-        return root.quantity_available
+    def resolve_stock_quantity(root: models.ProductVariant, info):
+        exact_quantity_available = root.quantity_available
+        if info.context.user.has_perm("product.manage_products"):
```

Comment:
We want this field to work the same for everyone. Admins have the `quantity` fields that they can rely on. `stockQuantity` however is meant for public API and it should behave in the same way for admins and for customers. So let's always return `min(exact_quantity_available, settings.MAX_CHECKOUT_LINE_QUANTITY)`.

Reply:
`stockQuantity` returns the stock available right now and `quantity` returns the stock available + allocated quantity. So there would be no way of knowing apart from `quantity - quantityAllocated`. We might want to introduce another field? `realStockQuantity` maybe as suggested previously?

Reply:
But we do have `quantityAllocated` which gives you this information don't we?